public/app/features/dashboard-scene/scene/RowRepeaterBehavior.ts

```diff
@@ -0,0 +1,215 @@
+        const sourceItemY = source.state.y ?? 0;
+        const itemY = sourceItemY + (rowContentHeight + 1) * index;
+
+        const itemClone = source.clone({
+          key: `${source.state.key}-clone-${index}`,
+          y: itemY,
+        });
+
+        //Make sure all the child scene objects have unique keys
+        ensureUniqueKeys(itemClone, index);
+
+        children.push(itemClone);
+
+        if (maxYOfRows < itemY + itemClone.state.height!) {
+          maxYOfRows = itemY + itemClone.state.height!;
+        }
+      }
+
+      const rowClone = this.getRowClone(rowToRepeat, index, values[index], texts[index], rowContentHeight, children);
+      rows.push(rowClone);
+    }
+
+    updateLayout(layout, rows, maxYOfRows, rowToRepeat);
+  }
+
+  getRowClone(
+    rowToRepeat: SceneGridRow,
+    index: number,
+    value: VariableValueSingle,
+    text: VariableValueSingle,
+    rowContentHeight: number,
+    children: SceneGridItemLike[]
+  ): SceneGridRow {
+    if (index === 0) {
+      rowToRepeat.setState({
+        // not activated
+        $variables: new SceneVariableSet({
+          variables: [new LocalValueVariable({ name: this.state.variableName, value, text: String(text) })],
+        }),
+        children,
+      });
+      return rowToRepeat;
+    }
+
+    const sourceRowY = rowToRepeat.state.y ?? 0;
+
+    return rowToRepeat.clone({
+      key: `${rowToRepeat.state.key}-clone-${index}`,
+      $variables: new SceneVariableSet({
+        variables: [new LocalValueVariable({ name: this.state.variableName, value, text: String(text) })],
+      }),
+      $behaviors: [],
+      children,
+      y: sourceRowY + rowContentHeight * index + index,
+    });
+  }
+}
+
+function getRowContentHeight(panels: SceneGridItemLike[]): number {
+  let maxY = 0;
+  let minY = Number.MAX_VALUE;
+
+  for (const panel of panels) {
+    if (panel.state.y! + panel.state.height! > maxY) {
+      maxY = panel.state.y! + panel.state.height!;
+    }
+    if (panel.state.y! < minY) {
+      minY = panel.state.y!;
+    }
+  }
+
+  return maxY - minY;
+}
+
+function updateLayout(layout: SceneGridLayout, rows: SceneGridRow[], maxYOfRows: number, rowToRepeat: SceneGridRow) {
+  const allChildren = getLayoutChildrenFilterOutRepeatClones(layout, rowToRepeat);
+  const index = allChildren.indexOf(rowToRepeat);
+
+  if (index === -1) {
+    throw new Error('RowRepeaterBehavior: Parent row not found in layout children');
+  }
+
+  const newChildren = [...allChildren.slice(0, index), ...rows, ...allChildren.slice(index + 1)];
+
+  // Is there grid items after rows?
+  if (allChildren.length > index + 1) {
+    const childrenAfter = allChildren.slice(index + 1);
+    const firstChildAfterY = childrenAfter[0].state.y!;
+    const diff = maxYOfRows - firstChildAfterY;
+
+    for (const child of childrenAfter) {
+      if (child.state.y! < maxYOfRows) {
+        child.setState({ y: child.state.y! + diff });
+      }
+    }
+  }
```

Comment:
This will be tricky potentially when we start working on edit mode. Do we need to update the y-positions? Thought react grid layout handles positioning if the full-width items push other items down.

Reply:
@dprokop we need to update yPos, the grid cannot handle this, it tries to but it will not know which row/grid item should be pushed down, so without this the repeated rows becomes mixed with non repeated rows, so if you have a row below the repeated row it will appear in the middle or in between repeated rows.

Reply:
I see, fair!